File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hm = HashMap::new();
        
        for i in 0..nums.len() {
            let needed = (target - nums[i]);
            
            if let Some(idx) = hm.get(&needed) {
               return vec![i as i32, *idx as i32];
            } else {
                hm.insert(nums[i], i);
            }
            
        }
        Vec::new()
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(0, len(nums)):
            for k in range(0, len(nums)):
                if i == k:
                    break
                if nums[i] + nums[k] == target:
                    return [i, k]